mig-agent/agent.go

```diff
@@ -457,6 +484,182 @@ func startRoutines(ctx *Context) (err error) {
 		ctx.Channels.Log <- mig.Log{Desc: "periodic environment refresh is disabled"}
 	}
 
+	// check for persistent modules, if enabled parse their config options
+	if len(PERSISTENTMODULES) > 0 {
+		for _, moduleString := range PERSISTENTMODULES {
+			moduleName := moduleString[0]
+			ctx.Channels.Log <- mig.Log{Desc: fmt.Sprintf("Enabled persistent module %q", moduleName)}
+			//parse moduleparams from the config file
+			//module params are passed as JSON MsgClassParameters Message
+			moduleParams := moduleString[1]
+			ctx.Channels.Log <- mig.Log{Desc: fmt.Sprintf("Feeding message to persistent module %q", moduleParams)}
+			// feed params to peristent module
+			err = startPersistentModule(ctx, moduleName, moduleParams)
```

Comment:
Is there a way we could validate the persistent module parameters are correctly formatted here (moduleParams) before we attempt to start it?